### PR TITLE
Enhancing the test for SG #3078

### PR DESCRIPTION
**Pull request by raghusarangapani**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:
- Enhancing the test test_channels_view_after_restart for SG #3078

### CODE REVIEW

File: testsuites/syncgateway/functional/tests/test_channels.py [hunk 1]
```diff
@@ -24,8 +24,11 @@ def test_channels_view_after_restart(params_from_base_test_setup, sg_conf_name):
     """
     - Add 10000 docs to Sync Gateway
     - Restart Sync Gateway (to flush channel cache)
+    - Add 1 doc
```
Comment (adamcfraser) [inline testsuites/syncgateway/functional/tests/test_channels.py]: To help people looking at this in the future, maybe change this to 
- Add 1 doc (to initialize the channel cache)

File: testsuites/syncgateway/functional/tests/test_channels.py [hunk 2]
```diff
@@ -73,6 +76,23 @@ def test_channels_view_after_restart(params_from_base_test_setup, sg_conf_name):
...
     sg.start_sync_gateways(cluster_config=cluster_config, url=sg_url, config=sg_conf)
 
+    # Add 1 to Sync Gateway
+    doc_bodies = document.create_docs(doc_id_prefix='doc_new', number=1, channels=seth_user_info.channels)
+    bulk_docs_resp_new = client.add_bulk_docs(url=sg_url, db=sg_db, docs=doc_bodies, auth=seth_session)
+    assert len(bulk_docs_resp_new) == 1
+
+    # Repopulate channel cache with view call
+    bulk_docs_resp += bulk_docs_resp_new
+    client.verify_docs_in_changes(url=sg_url, db=sg_db, expected_docs=bulk_docs_resp, auth=seth_session)
```
Comment (adamcfraser) [inline testsuites/syncgateway/functional/tests/test_channels.py]: expected_docs should now be bulk_docs_resp + bulk_docs_resp_new, right?

File: testsuites/syncgateway/functional/tests/test_channels.py [hunk 3]
```diff
@@ -73,6 +76,23 @@ def test_channels_view_after_restart(params_from_base_test_setup, sg_conf_name):
...
+
+    # Get Sync Gateway Expvars
+    expvars = client.get_expvars(url=sg_admin_url)
+
+    # Only check the view querys if in channel cache mode
+    if mode == 'cc':
+        log_info('Looking for view queries == 1 in expvars')
+        assert expvars['syncGateway_changeCache']['view_queries'] == 1
+
     # Repopulate channel cache with view call
```
Comment (adamcfraser) [inline testsuites/syncgateway/functional/tests/test_channels.py]: Update the comment on line 96 to say 'issue a second changes request that shouldn't trigger a view call'

File: testsuites/syncgateway/functional/tests/test_channels.py [hunk 4]
```diff
@@ -73,6 +76,23 @@ def test_channels_view_after_restart(params_from_base_test_setup, sg_conf_name):
     sg.stop_sync_gateways(cluster_config=cluster_config, url=sg_url)
     sg.start_sync_gateways(cluster_config=cluster_config, url=sg_url, config=sg_conf)
 
+    # Add 1 to Sync Gateway
+    doc_bodies = document.create_docs(doc_id_prefix='doc_new', number=1, channels=seth_user_info.channels)
+    bulk_docs_resp_new = client.add_bulk_docs(url=sg_url, db=sg_db, docs=doc_bodies, auth=seth_session)
+    assert len(bulk_docs_resp_new) == 1
+
+    # Repopulate channel cache with view call
```
Comment (adamcfraser) [inline testsuites/syncgateway/functional/tests/test_channels.py]: Maybe more descriptive to say "Changes request to trigger population of channel cache with view call"

File: testsuites/syncgateway/functional/tests/test_channels.py (hunk 1 above)
Comment (raghusarangapani) [reply testsuites/syncgateway/functional/tests/test_channels.py]: Done

File: testsuites/syncgateway/functional/tests/test_channels.py (hunk 4 above)
Comment (raghusarangapani) [reply testsuites/syncgateway/functional/tests/test_channels.py]: Done

File: testsuites/syncgateway/functional/tests/test_channels.py (hunk 2 above)
Comment (raghusarangapani) [reply testsuites/syncgateway/functional/tests/test_channels.py]: Yes, lone 85 is `bulk_docs_resp += bulk_docs_resp_new`

File: testsuites/syncgateway/functional/tests/test_channels.py (hunk 3 above)
Comment (raghusarangapani) [reply testsuites/syncgateway/functional/tests/test_channels.py]: Done

**Pull request merged**